Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Сreate/CreateRelatedFigureHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using AutoMapper;
+using FluentResults;
+using MediatR;
+using NLog.Targets;
+using Streetcode.DAL.Entities.Streetcode;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+
+namespace Streetcode.BLL.MediatR.Streetcode.RelatedFigure.Create;
+
+public class CreateRelatedFigureHandler : IRequestHandler<CreateRelatedFigureCommand, Result<Unit>>
+{
+    private readonly IMapper _mapper;
+    private readonly IRepositoryWrapper _repositoryWrapper;
+
+    public CreateRelatedFigureHandler(IRepositoryWrapper repositoryWrapper, IMapper mapper)
+    {
+        _repositoryWrapper = repositoryWrapper;
+        _mapper = mapper;
+    }
+
+    public async Task<Result<Unit>> Handle(CreateRelatedFigureCommand request, CancellationToken cancellationToken)
+    {
+        var entity1 = await _repositoryWrapper.StreetcodeRepository.GetFirstOrDefaultAsync(rel => rel.Id == request.ObserverId);
+        var entity2 = await _repositoryWrapper.StreetcodeRepository.GetFirstOrDefaultAsync(rel => rel.Id == request.TargetId);
+
+        if (entity1 is null)
+        {
+            return Result.Fail(new Error($"No existing streetcode with id: {request.ObserverId}"));
+        }
+
+        if (entity2 is null)
+        {
+            return Result.Fail(new Error($"No existing streetcode with id: {request.TargetId}"));
+        }
```

Comment:
You should not reapet yourself. You could create one if condition like this:
if(entity1 is null || entity2 is null) {
...
}

Also names of entity1 and entity2 could be misunderstood. Maybe rename them as entityObserved and entityTarget.

Reply:
i need to know which value is null to return a proper response